node/lib/util/synthetic_branch_util.js

```diff
@@ -157,6 +158,41 @@ function* checkSubmodule(repo, cfg, metaCommit, submoduleEntry, url) {
     }
 }
 
+/**
+ * Return a list of submodules changed or added in between `commit`
+ * and `parent`.  Exclude deleted submodules.
+ *
+ * If parent is null, return null.
+ */
+function* computeChangedSubmodules(repo, commit, parent) {
```

Comment:
You might consider using `SubmoduleUtil.getSubmoduleChanges`. It does almost exactly the same thing.